### DISCUSSION TITLE
Question about settings

### DISCUSSION THREAD

**Discussion by PiousVenom** [PHP-CS-Fixer/PHP-CS-Fixer#7964] [Q&A] [answered]
I have the following loop (this is for loading the relationships of specific models):

```php
        foreach($this as $order) {
            /** @var Order $order */
            $order->load('merchant');
            $order->load('payments');
        }
```

Whenever I run the fixer, it updates to this:

```php
        foreach ($this as $order) {
            // @var Order $order
            $order->load('merchant');
            $order->load('payments');
        }
```

What can I do to keep the doc block as is? After running the fixer and then running phpstan errors (can't call load() on mixed).

Thanks for the help!

**Comment by Wirone** [comment, marked as answer]
You should provide proper internal types to the class that you iterate over, so type of the `$order` is inferred automatically, and known to IDE and PHPStan. It probably requires some generics and other stuff. In general, the `@var Order $order` should not be needed there. It's more about your code's architecture than Fixer config.

**Reply by keradus** [reply]
one of possibility: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/src/Tokenizer/Tokens.php#L30
